runtime/backtrace.c

```diff
@@ -136,34 +136,22 @@ CAMLprim value caml_get_exception_raw_backtrace(value unit)
   CAMLparam0();
   CAMLlocal1(res);
 
-  /* Beware: the allocations below may cause finalizers to be run, and another
-     backtrace---possibly of a different length---to be stashed (for example
-     if the finalizer raises then catches an exception).  We choose to ignore
-     any such finalizer backtraces and return the original one. */
-
   if (!caml_backtrace_active ||
       caml_backtrace_buffer == NULL ||
       caml_backtrace_pos == 0) {
     res = caml_alloc(0, 0);
   }
   else {
-    backtrace_slot saved_caml_backtrace_buffer[BACKTRACE_BUFFER_SIZE];
-    int saved_caml_backtrace_pos;
     intnat i;
 
-    saved_caml_backtrace_pos = caml_backtrace_pos;
-
-    if (saved_caml_backtrace_pos > BACKTRACE_BUFFER_SIZE) {
-      saved_caml_backtrace_pos = BACKTRACE_BUFFER_SIZE;
-    }
-
-    memcpy(saved_caml_backtrace_buffer, caml_backtrace_buffer,
-           saved_caml_backtrace_pos * sizeof(backtrace_slot));
-
-    res = caml_alloc(saved_caml_backtrace_pos, 0);
-    for (i = 0; i < saved_caml_backtrace_pos; i++) {
-      Field(res, i) = Val_backtrace_slot(saved_caml_backtrace_buffer[i]);
+    /* We use caml_alloc_shr instead of caml_alloc, since caml_alloc might
```

Comment:
All these changes in backtrace.c are unrelated to this PR, right?

Reply:
They're in a seperate commit, but a related one. (It may be easier to read the PR commit-by-commit).

One of the changes to fix backtraces is to lower the stack probe amount from 32k to 4k, since a 32k stack probe is not reliable (on my machine, glibc pthreads uses a single 4k guard page, and allocates memory just afterwards for other purposes, so a 32k stack probe has false positives).

Only having 4k of stack guaranteed means that it is unwise to allocate 8k arrays on the stack, especially in functions that are likely to be called from a `Stack_overflow` handler. `caml_get_exception_raw_backtrace` was such a function, and the second commit in this PR fixes it.